modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/VersionResolver.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import static org.mule.config.i18n.MessageFactory.createStaticMessage;
+import org.mule.api.MuleRuntimeException;
+import org.mule.extension.annotations.Extension;
+
+/**
+ * This class infers the version of an extension from the MANIFEST.MF contained in it's JAR.
+ * If this cannot be accomplished, a fallback method is executed.
+ */
+public abstract class VersionResolver
+{
+
+    public String resolveVersion(Class extensionType, Extension extension)
```

Comment:
This method should not receive the extensionType. Although the default, introspecting annotations on a class is just one way  to use the ext-api. This should only receive the Extension and if it needs the extensionType that should be something a  particular implementation takes on the constructor